react-ui/src/applyFilter.js

```diff
@@ -10,19 +12,46 @@ export default async function applyFilter(imageData, filter) {
     canvas = canvas || document.createElement('canvas')
 
     const image = new Image()
-    image.onload = () => {
+    image.onload = async () => {
       const ctx = canvas.getContext('2d')
 
       canvas.width = image.width
       canvas.height = image.height
       ctx.drawImage(image, 0, 0, image.width, image.height)
 
       const imgData = ctx.getImageData(0, 0, canvas.width, canvas.height)
-      const withFilter = window.pixelsJS.filterImgData(imgData, filter)
 
-      ctx.putImageData(withFilter, 0, 0)
+      // Choose whether to use pixelsJS or styleTransfer
+      var is_style = (typeof filter != 'string')
+      var orig_data = imgData.data.slice(0)
+      let transformedImg
+      if (is_style) {
+        var style = filter
+
+        if (weight<1.0) { // Linear interpolation between base and given style
+          base_style = base_style || await getImageStyle(imgData)
+          style = linearCombination(base_style,style,weight)
+        }
+
+        transformedImg = await transformImageWithStyle(imgData, style)
+      }
+      else
+        transformedImg = window.pixelsJS.filterImgData(imgData, filter)
+
+      if (weight<1.0) { // Linear interpolation between pixels of filter and original
```

Comment:
Usually comments like this are an indication that a function should be extracted.